src/Equinor.ProCoSys.IPO.Infrastructure/Repositories/InvitationRepository.cs

```diff
@@ -134,5 +134,50 @@ public void RemoveInvitation(Invitation invitation)
             }
             _context.Invitations.Remove(invitation);
         }
+
+        public void UpdateRfocStatuses(string projectName, IList<string> commPkgNos, IList<Tuple<string, string>> mcPkgs)
+        {
+            var project = _context.Projects.SingleOrDefault(x => x.Name.Equals(projectName));
+
+            var invitations = _context.Invitations
+                .Include(i => i.McPkgs)
+                .Include(i => i.CommPkgs)
+                .Where(i => i.ProjectId == project.Id
+                            && (i.CommPkgs.Any(x => commPkgNos.Any(y => y == x.CommPkgNo))
+                            || i.McPkgs.Any(x => mcPkgs.Any(y => y.Item1 == x.McPkgNo && y.Item2 == x.CommPkgNo))))
+                .ToList();
+
+            foreach (var invitation in invitations)
+            {
+                if (invitation.Type == DisciplineType.MDP)
+                {
+                    UpdateRfocStatusForMDP(invitation, commPkgNos);
```

Comment:
UpdateRfocStatusForMDP does more than just update the status. Consider moving scope  logic out of the method and call it after?